service/service.go

```diff
@@ -1873,3 +1882,21 @@ func ParseInt64FromContext(ctx context.Context, key string) (int64, error) {
 func lookupEnv(ctx context.Context, key string) (string, bool) {
 	return csictx.LookupEnv(ctx, key)
 }
+
+func getZoneKeyLabelFromSecret(arrays map[string]*ArrayConnectionData) (string, error) {
+	zoneKeyLabel := ""
+
+	for _, array := range arrays {
+		if array.AvailabilityZone != nil {
+			if zoneKeyLabel == "" {
+				// Assumes that the key parameter is not empty
+				zoneKeyLabel = array.AvailabilityZone.LabelKey
```

Comment:
**question:** What if the `labelKey` is left empty? Should we default to `topology.kubernetes.io/zone`?

Reply:
Initially, I considered setting a default value. However, since the customer needs to explicitly label their nodes with the zone label, they should also be able to specify that label in the Secret. Therefore, I decided not to include a default label.